diff --git a/bmp/bmpfileheader.cpp b/bmp/bmpfileheader.cpp
--- a/bmp/bmpfileheader.cpp
+++ b/bmp/bmpfileheader.cpp
@@ -3,7 +3,7 @@
 #include "bmpfileheader.h"
-BmpFileHeader::BmpFileHeader() {
+BmpFileHeader::BmpFileHeader(const BmpFileHeader& newBf) : bfType(newBf.bfType), bfSize(newBf.bfSize), bfReserved1(newBf.bfReserved1), bfReserved2(newBf.bfReserved2), bfOffBits(newBf.bfOffBits) {}
 BmpFileHeader::BmpFileHeader(int NZwidth, int height) {
     // Initialize uninit member vars here
     bfType = 1; // Assuming a default value for bfType
     bfSize = NZwidth * height * 3; // Estimate size based on pixel format
     bfReserved1 = 0;
     bfReserved2 = 0;
     bfOffBits = sizeof(BmpFileHeader) + (NZwidth * height * 3);
 }