diff --git a/bmp/bmpinfoheader.cpp b/bmp/bmpinfoheader.cpp
--- a/bmp/bmpinfoheader.cpp
+++ b/bmp/bmpinfoheader.cpp
@@ -3,9 +3,9 @@
 #include "bmpinfoheader.h"
-BmpInfoHeader::BmpInfoHeader() {
-    // Uninitialized member variables
+BmpInfoHeader::BmpInfoHeader(const BmpFileHeader& newBf, int NZwidth, int height) {
+    bfType = newBf.bfType;
+    bfSize = newBf.bfSize;
+    bfReserved1 = newBf.bfReserved1;
+    bfReserved2 = newBf.bfReserved2;
+    bfOffBits = newBf.bfOffBits;
+    biSize = sizeof(BmpInfoHeader);
+    biWidth = NZwidth;
+    biHeight = height;
+    // Initialize remaining member variables as needed
 }