diff --git a/bmp/bmpinfoheader.cpp b/bmp/bmpinfoheader.cpp
--- a/bmp/bmpinfoheader.cpp
+++ b/bmp/bmpinfoheader.cpp
@@ -1,11 +1,12 @@
-BmpInfoHeader::BmpInfoHeader() {
-    // Uninitialized member variables
+BmpInfoHeader::BmpInfoHeader(int NZwidth, int height) :
+    biSize(0), biWidth(0), biHeight(0), biPlanes(0), biBitCount(0),
+    biCompression(0), biSizeImage(0), biXPelsPerMeter(0),
+    biYPelsPerMeter(0), biClrUsed(0), biClrImportant(0) {
+    // Initialize member variables here