diff --git a/bmp/bmpfileheader.cpp b/bmp/bmpfileheader.cpp
--- a/bmp/bmpfileheader.cpp
+++ b/bmp/bmpfileheader.cpp
@@ -1,10 +1,14 @@
-BmpFileHeader::BmpFileHeader()
+BmpFileHeader::BmpFileHeader(int NZwidth, int height, BmpFileHeader& newBf, BmpInfoHeader& newBi, std::vector<uint8_t> img)
    // Initialize member variables in the constructor
    bfType = newBf.bfType;
    bfSize = newBf.bfSize;
    bfReserved1 = newBf.bfReserved1;
    bfReserved2 = newBf.bfReserved2;
    bfOffBits = newBf.bfOffBits;
    // Assign remaining constructor arguments
    biSize = newBi.biSize;
    biWidth = newBi.biWidth;
    biHeight = newBi.biHeight;
    biPlanes = newBi.biPlanes;
    biBitCount = newBi.biBitCount;
    biCompression = newBi.biCompression;
    biSizeImage = newBi.biSizeImage;
    biXPelsPerMeter = newBi.biXPelsPerMeter;
    biYPelsPerMeter = newBi.biYPelsPerMeter;
    biClrUsed = newBi.biClrUsed;
    biClrImportant = newBi.biClrImportant;