diff --git a/bmp/bmp.cpp b/bmp/bmp.cpp
--- a/bmp/bmp.cpp
+++ b/bmp/bmp.cpp
@@ -12,12 +12,12 @@
     BmpFileHeader newBf(NZwidth, height);
     BmpInfoHeader newBi(NZwidth, height);
-    bf = newBf;
-    bi = newBi;
-    imgData = img;
+    // Initialize variables in initialization lists
+    Bmp::Bmp(bf, newBi, img);
 void Bmp::setImgData(std::vector<uint8_t> newImgData) {
     imgData = newImgData;
 }