diff --git a/bmp/bmpinfoheader.cpp b/bmp/bmpinfoheader.cpp
--- a/bmp/bmpinfoheader.cpp
+++ b/bmp/bmpinfoheader.cpp
@@ -1,7 +1,7 @@
-BmpInfoHeader::BmpInfoHeader() {
+BmpInfoHeader::BmpInfoHeader(int NZwidth, int height) : biWidth(NZwidth), biHeight(height) {}
-BmpInfoHeader::BmpInfoHeader() : biSize(0), biWidth(0), biHeight(0) {}
+// Initialize all member variables in the initialization list
+BmpInfoHeader::BmpInfoHeader(int NZwidth, int height) :
+    biSize(sizeof(BmpInfoHeader)), biWidth(NZwidth), biHeight(height) {}