diff --git a/bmp/bmpinfoheader.cpp b/bmp/bmpinfoheader.cpp
--- a/bmp/bmpinfoheader.cpp
+++ b/bmp/bmpinfoheader.cpp
@@ -1,12 +1,12 @@
-BmpInfoHeader::BmpInfoHeader() {
-    // Uninitialized member variables
+BmpInfoHeader::BmpInfoHeader(const BmpFileHeader& newBf, const BmpInfoHeader& newBi, std::vector<uint8_t> img) :
+    bfType(newBf.bfType),
+    bfSize(newBf.bfSize),
+    bfReserved1(newBf.bfReserved1),
+    bfReserved2(newBf.bfReserved2),
+    bfOffBits(newBf.bfOffBits),
+    biSize(newBi.biSize),
+    biWidth(newBi.biWidth),
+    biHeight(newBi.biHeight),
+    biPlanes(newBi.biPlanes),
+    biBitCount(newBi.biBitCount),
+    biCompression(newBi.biCompression),
+    biSizeImage(newBi.biSizeImage),
+    biXPelsPerMeter(0),
+    biYPelsPerMeter(0),
+    biClrUsed(0),
+    biClrImportant(0) {
+    // Initialize remaining members