diff --git a/bmp/bmp.cpp b/bmp/bmp.cpp
--- a/bmp/bmp.cpp
+++ b/bmp/bmp.cpp
@@ -12,12 +12,12 @@
     BmpFileHeader newBf(NZwidth, height);
     BmpInfoHeader newBi(NZwidth, height);
-    bf = newBf;         // Change to initialization list
-    bi = newBi;         // Change to initialization list
-    imgData = img;       // Change to initialization list
+    Bmp::Bmp(newBf, newBi, img);  // Move initialization to constructor
 void Bmp::setImgData(const std::vector<uint8_t>& newImgData) {
     imgData = newImgData;
 }