diff --git a/bmp/bmpinfoheader.cpp b/bmp/bmpinfoheader.cpp
--- a/bmp/bmpinfoheader.cpp
+++ b/bmp/bmpinfoheader.cpp
@@ -1,12 +1,14 @@
-BmpInfoHeader::BmpInfoHeader()
+BmpInfoHeader::BmpInfoHeader(int NZwidth, int height) :
     biSize(0), biWidth(0), biHeight(0), biPlanes(1), biBitCount(24), biCompression(0),
     biSizeImage(0), biXPelsPerMeter(0), biYPelsPerMeter(0), biClrUsed(0), biClrImportant(0)
    // Initialize member variables
    biSize = sizeof(BmpInfoHeader);
    biWidth = NZwidth;
    biHeight = height;