diff --git a/puzzle-challenge/labels.py b/puzzle-challenge/labels.py
--- a/puzzle-challenge/labels.py
+++ b/puzzle-challenge/labels.py
@@ -132,7 +132,7 @@
         self.clicked = False
         self.hover = False
     def render_text(self):
-        font, size = self.font, self.font_size
+        font, size = self.font_size, self.font_size
         if (font, size) not in LOADED_FONTS:
             LOADED_FONTS[font, size] = pg.font.Font(font, size)
         self.font = LOADED_FONTS[font, size]
         text = self.text and self.font.render(self.text, 1, self.text_color)
         hover = self.hover_text and self.font.render(self.hover_text, 1,